gateway-ha/src/test/java/io/trino/gateway/ha/router/TestTrinoRequestUser.java

```diff
@@ -42,4 +52,34 @@ void testJsonCreator()
         assertThat(deserializedTrinoRequestUser.getUser()).isEqualTo(trinoRequestUser.getUser());
         assertThat(deserializedTrinoRequestUser.getUserInfo()).isEqualTo(trinoRequestUser.getUserInfo());
     }
+
+    @Test
+    void testUserFromJwtToken()
+    {
+        final String USER_NAME_CLAIM = "username";
+        final String USER_NAME_VALUE = "trino";
+
+        RequestAnalyzerConfig requestAnalyzerConfig = new RequestAnalyzerConfig();
+        requestAnalyzerConfig.setTokenUserField(USER_NAME_CLAIM);
+
+        Algorithm algorithm = Algorithm.HMAC256("random");
+
+        Instant expiryTime =  Instant.now().plusSeconds(60);
+        String token = JWT.create()
+                .withIssuer("gateway")
+                .withClaim(USER_NAME_CLAIM, USER_NAME_VALUE)
+                .withExpiresAt(Date.from(expiryTime))
+                .sign(algorithm);
+
+        HttpServletRequest mockRequest = mock(HttpServletRequest.class);
+        when(mockRequest.getHeader(USER_HEADER)).thenReturn(null);
+        when(mockRequest.getHeader(HttpHeaders.AUTHORIZATION)).thenReturn("Bearer " + token);
+
+        TrinoRequestUser trinoRequestUser = new TrinoRequestUser
+                .TrinoRequestUserProvider(requestAnalyzerConfig).getInstance(mockRequest);
+
+        assertThat(trinoRequestUser.getUser().isPresent()).isEqualTo(true);
+        assertThat(trinoRequestUser.getUser().orElseThrow()).isEqualTo(USER_NAME_VALUE);
```

Comment:
Please avoid boolean assertion as much as possible as the failure message is unhelpful. `hasValue` is sufficient in this case:
```suggestion
        assertThat(trinoRequestUser.getUser()).hasValue(USER_NAME_VALUE);
```